web/mass_class/consumers.py

```
@@ -0,0 +1,517 @@
+                }
+            )
+        )
+
+    async def disconnect(self, close_code):
+        # Leave room group
+        await self.channel_layer.group_discard(self.room_group_name, self.channel_name)
+
+        # Close any active stream
+        await database_sync_to_async(self._close_stream)(self.session_id)
+
+    async def receive(self, text_data):
```

Ruff (0.8.2)

Missing return type annotation for public function `receive`

Add return type annotation: `None`

(ANN201)